src/savegame.cpp

```diff
@@ -256,75 +223,54 @@ bool loadgame::check_version_compatibility(const version_info& save_version)
 	return true;
 }
 
-void loadgame::set_gamestate()
+// TODO: reduce code duplication with load_interactive
+utils::optional<load_game_metadata> load_interactive_for_multiplayer()
 {
-	gamestate_.set_data(load_data_.load_config);
-}
-
-bool loadgame::load_multiplayer_game()
-{
-	if(!gui2::dialogs::game_load::execute(game_config_, load_data_)) {
-		return false;
-	}
+	// FIXME: game_load dialog should initialize its own manager pointer
+	load_game_metadata load_data{ save_index_class::default_saves_dir() };
 
-	load_data_.show_replay |= is_replay_save(load_data_.summary);
-	if(load_data_.filename.empty()) {
-		return false;
+	if(!gui2::dialogs::game_load::execute(load_data)) {
+		return utils::nullopt;
 	}
 
-	if(!load_data_.manager) {
-		ERR_SAVE << "Null pointer in save index";
-		return false;
-	}
+	load_data.show_replay |= is_replay_save(load_data.summary);
 
 	// read_save_file needs to be called before we can verify the classification so the data has
 	// been populated. Since we do that, we report any errors in that process first.
 	try {
 		cursor::setter cur(cursor::WAIT);
 		log_scope("load_game");
 
-		load_data_.load_config = read_save_file(load_data_.manager->dir(), load_data_.filename);
+		load_data.load_config = read_save_file(load_data.manager->dir(), load_data.filename);
 	} catch(const game::load_game_failed& e) {
 		gui2::show_error_message(_("The file you have tried to load is corrupt") + "\n\n" + e.what());
-		return false;
+		return utils::nullopt;
 	}
 
-	if(is_replay_save(load_data_.summary)) {
+	if(is_replay_save(load_data.summary)) {
 		gui2::show_transient_message(_("Load Game"), _("Replays are not supported in multiplayer mode."));
-		return false;
+		return utils::nullopt;
 	}
 
-	// Since we want to verify game classification and version compatibility before setting the
-	// gamestate, we can't make use of the game_classification object owned by the gamestate.
-	const auto metadata = game_classification{load_data_.load_config};
+	const auto metadata = game_classification{load_data.load_config};
 
 	if(!metadata.is_multiplayer()) {
 		gui2::show_transient_error_message(_("This is not a multiplayer save."));
-		return false;
-	}
-
-	copy_era(load_data_.load_config);
-	return check_version_compatibility(metadata.version);
-}
-
-void loadgame::copy_era(config& cfg)
-{
-	auto replay_start = cfg.optional_child("replay_start");
-	if(!replay_start) {
-		return;
+		return utils::nullopt;
 	}
 
-	auto era = replay_start->optional_child("era");
-	if(!era) {
-		return;
+	if(!check_version_compatibility(metadata.version)) {
+		return utils::nullopt;
 	}
 
-	auto snapshot = cfg.optional_child("snapshot");
-	if(!snapshot) {
-		return;
-	}
+	copy_era(load_data.load_config);
```

Comment:
i know its not really caused by your changed, but i'm really curious why we are copying the era here, or rather in which cases the [era] in [replay_start] exists and why it is needed.

Reply:
The code in question has been around since 2009, so it’s quite possibly dead. 9fabc334cb70b8cbbd235c456fc3916293f15f82